tests/unit/strictdoc/backend/sdoc/test_dsl_passthrough.py

```diff
@@ -945,3 +955,357 @@ def test_153_grammar_tag():
+
+    assert sdoc_input == output
+
+
+def test_161_grammar_refs_file():
+    sdoc_input = """
+[DOCUMENT]
+TITLE: Test Doc
+
+[GRAMMAR]
+ELEMENTS:
+- TAG: LOW_LEVEL_REQUIREMENT
+  FIELDS:
+  - TITLE: REFS
+    TYPE: TypeValue(ParentReqReference, FileReference)
+    REQUIRED: False
+
+[LOW_LEVEL_REQUIREMENT]
+REFS:
+- TYPE: File
+  VALUE: /tmp/sample.cpp
+""".lstrip()
+
+    reader = SDReader()
+
+    document = reader.read(sdoc_input)
+    assert isinstance(document, Document)
+
+    ll_requirement = document.section_contents[0]
+    references = ll_requirement.references
+    assert len(references) == 1
+
+    reference = references[0]
+    assert isinstance(reference, FileReference)
+    assert reference.ref_type == ReferenceType.FILE
+    assert reference.path == "/tmp/sample.cpp"
+
+    writer = SDWriter()
+    output = writer.write(document)
+
+    assert sdoc_input == output
+
+
+def test_162_grammar_refs_file_multi():
+    sdoc_input = """
+[DOCUMENT]
+TITLE: Test Doc
+
+[GRAMMAR]
+ELEMENTS:
+- TAG: LOW_LEVEL_REQUIREMENT
+  FIELDS:
+  - TITLE: REFS
+    TYPE: TypeValue(ParentReqReference, FileReference)
+    REQUIRED: False
+
+[LOW_LEVEL_REQUIREMENT]
+REFS:
+- TYPE: File
+  VALUE: /tmp/sample1.cpp
+- TYPE: File
+  VALUE: /tmp/sample2.cpp
+- TYPE: File
+  VALUE: /tmp/sample3.cpp
+""".lstrip()
+
+    reader = SDReader()
+
+    document = reader.read(sdoc_input)
+    assert isinstance(document, Document)
+
+    ll_requirement = document.section_contents[0]
+    references = ll_requirement.references
+    assert len(references) == 3
+
+    reference = references[0]
+    assert isinstance(reference, FileReference)
+    assert reference.ref_type == ReferenceType.FILE
+    assert reference.path == "/tmp/sample1.cpp"
+
+    reference = references[1]
+    assert isinstance(reference, FileReference)
+    assert reference.ref_type == ReferenceType.FILE
+    assert reference.path == "/tmp/sample2.cpp"
+
+    reference = references[2]
+    assert isinstance(reference, FileReference)
+    assert reference.ref_type == ReferenceType.FILE
+    assert reference.path == "/tmp/sample3.cpp"
+
+    writer = SDWriter()
+    output = writer.write(document)
+
+    assert sdoc_input == output
+
+
+def test_163_grammar_refs_file_only():
+    sdoc_input = """
+[DOCUMENT]
+TITLE: Test Doc
+
+[GRAMMAR]
+ELEMENTS:
+- TAG: LOW_LEVEL_REQUIREMENT
+  FIELDS:
+  - TITLE: UID
+    TYPE: String
+    REQUIRED: True
+  - TITLE: REFS
+    TYPE: TypeValue(FileReference)
+    REQUIRED: False
+
+[LOW_LEVEL_REQUIREMENT]
+UID: ID-001
+
+[LOW_LEVEL_REQUIREMENT]
+UID: ID-002
+REFS:
+- TYPE: Parent
+  VALUE: ID-001
+- TYPE: File
+  VALUE: /tmp/sample1.cpp
+""".lstrip()
+
+    reader = SDReader()
+
+    try:
+        document = reader.read(sdoc_input)
+        assert isinstance(document, Document)
+
+        ll_requirement = document.section_contents[0]
+        references = ll_requirement.references
+        assert len(references) == 0
+
+    except Exception as err:
```

Comment:
Writing the test with try/except, you make the test always pass. If you want to test the positive outcome, remove try/except. If you wish to test that the exception is thrown, rather use this example and write an independent test case for error handling: `test_003_one_range_pragma_begin_req_not_equal_to_end_req`.

Reply:
Changed the UnitTests test_dsl_passthrough.py
def test_163_grammar_refs_file_only()
def test_166_grammar_refs_parent_only():
into `test that the exception is thrown` variant.